python/tvm/relax/ir/instrument.py

```diff
@@ -25,13 +25,19 @@ class WellFormedInstrument:
     is well formed. It will skip specific passes, like Normalize.
     """
 
-    def __init__(self):
+    def __init__(self, check_struct_info=True):
         self.skip_pass_name = ["Normalize", "ResolveGlobals"]
+        self.check_struct_info = check_struct_info
 
     def run_before_pass(self, mod, pass_info):
-        if pass_info.name not in self.skip_pass_name:
-            assert relax.analysis.well_formed(mod)
+        self._check(mod, pass_info.name, "Before")
 
     def run_after_pass(self, mod, pass_info):
-        if pass_info.name not in self.skip_pass_name:
-            assert relax.analysis.well_formed(mod)
+        self._check(mod, pass_info.name, "After")
```

Comment:
Nit: Do we need in both before and after? If we have consecutive passes, we would have redundant checks running.

Reply:
When doing an end-to-end lowering, it's definitely overkill, but this utility is primarily used for the relax unit tests.  There, I think it's helpful in establishing whether a transformation caused a module to be ill-formed, or whether the usually-hand-written input module was ill-formed from the start.

Reply:
got it. yeah, for testing purpose, I think it is okay. we can revisit in the future if necessary.